### DISCUSSION TITLE
How do SwitchTopic 1 and 2 work?

### DISCUSSION THREAD

**Discussion by AndreKR** [arendst/Tasmota#14656] [Q&A] [answered]
So `SwitchTopic` has 4 possible values:

* `0`: The default. The switch controls the relay and also sends MQTT messages on `stat/%topic/POWER`.
* `1`: The switch now also sends on `cmnd/%topic%/POWER1`. The switch still controls the relay, possibly because it sees its own messages, but the result is still the same.
* `2`: Works exactly the same as `SwitchTopic 0`, only that the command prints `{"SwitchTopic":"0"}` instead of `{"SwitchTopic":""}` as result.
* `<custom value>`: This sends MQTT messages on `cmnd/<custom value>/POWER1`.

`0` and `<custom value>` make sense to me, but `1` and `2` do not.

**Comment by barbudor** [comment, marked as answer]
From the documetation, your interpretation of `0` is wrong:
> 0 = disable use of MQTT switch topic
By default a switch controls the corresponding power state and doesn't send any MQTT messages itself.
No MQTT message will be published on account of the new switch state. The message you see in console is the new power state that is controlled and not the switch state.

Same for `2` which reverts to `MQTT_SWITCH_TOPIC` 
As `MQTT_SWITCH_TOPIC` can be overriden in a custom build, it is not necesseraly the same as `0`

For `1`, there must be a use case as this has been implemented.

**Reply by AndreKR** [reply]
Oh, so now I get what that `0` means... `SwitchTopic 2` basically runs the command `SwitchTopic <MQTT_SWITCH_TOPIC>` and by default that is `0` so it behaves identical to `SwitchTopic 0` unless `MQTT_SWITCH_TOPIC` was overridden. :bulb:

But `1` still puzzles me. Okay, with `0` the power state is manipulated internally and as a result an MQTT message is published. With `1` the power state is manipulated via MQTT? So what, the result is still the same.

**Comment by maximr377** [comment]
Hello! Sorry for the question but I'm a newbie and I'm getting dizzy. I want to control a device with a relay from another that only has buttons and I don't know how to send the commands to turn it on through mqqt. Might you help me?

**Reply by barbudor** [reply]
On the device with the buttons, you can set the SwitcTopic to the topic to be controlled
